cmd/virt-handler/virt-handler.go

```diff
@@ -73,7 +74,13 @@ func main() {
 	broadcaster.StartRecordingToSink(&kubecorev1.EventSinkImpl{Interface: coreClient.Events(api.NamespaceDefault)})
 	recorder := broadcaster.NewRecorder(kubev1.EventSource{Component: "virt-handler", Host: *host})
 
-	domainManager, err := virtwrap.NewLibvirtDomainManager(domainConn, recorder)
+	var domainManager virtwrap.DomainManager
+	switch *driver {
+	case "ovirt":
+		domainManager, err = virtwrap.NewOvirtDomainManager(domainConn, recorder)
+	default:
+		domainManager, err = virtwrap.NewLibvirtDomainManager(domainConn, recorder)
+	}
```

Comment:
@admiyo 

> What this is showing is the need for a driver architecture for the VM launch mechanism. VDSM can be viewed as a wrapper around libvirt that provides needed functionality that cannot be expressed in libvirt. Thus, saying that libvirt is the one-and-only-true VM launch mechanism is too harsh a stance.

I see it the same way.

> As I see it, we have two choices: drivers in tree and drivers out of tree. Since I suspect we will have a vanishingly few number of drivers, I suggest we pull them into the tree for now.

Sounds good. No one knows if we will have such drivers on the long run, so being open for them but not investing too much in infra for them for now, looks like a good approach to me.